src/main/java/at/hannibal2/skyhanni/features/misc/FrogMaskFeatures.kt

```diff
@@ -0,0 +1,141 @@
+import at.hannibal2.skyhanni.config.ConfigUpdaterMigrator
+import at.hannibal2.skyhanni.config.features.misc.frogmask.FrogMaskWarningConfig.WarningType
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.data.ScoreboardData
+import at.hannibal2.skyhanni.data.TitleManager
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.SecondPassedEvent
+import at.hannibal2.skyhanni.features.skillprogress.SkillProgress.updateSkillInfo
+import at.hannibal2.skyhanni.features.skillprogress.SkillType
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.getInternalName
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.NeuInternalName.Companion.toInternalName
+import at.hannibal2.skyhanni.utils.NeuItems.getItemStack
+import at.hannibal2.skyhanni.utils.RegexUtils.firstMatcher
+import at.hannibal2.skyhanni.utils.RenderUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderable
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SimpleTimeMark.Companion.asTimeMark
+import at.hannibal2.skyhanni.utils.SkyBlockTime
+import at.hannibal2.skyhanni.utils.TimeUtils.format
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object FrogMaskFeatures {
+    private val config get() = SkyHanniMod.feature.misc.frogMaskFeatures
+
+    private var display: Renderable? = null
+
+    private var lastWarning = SimpleTimeMark.farPast()
+
+    private val patternGroup = RepoPattern.group("misc.frogmask")
+
+    /**
+     * REGEX-TEST: §7Today's region: §aDark Thicket
+     */
+    private val activeRegionPattern by patternGroup.pattern(
+        "description.active",
+        "§7Today's region: (?<region>.+)",
+    )
+
+    /**
+     * REGEX-TEST:  §7⏣ §aSpruce Woods
+     */
+    private val currentAreaPattern by patternGroup.pattern(
+        "scoreboard.current",
+        " §7⏣ (?<area>.+)",
+    )
+
+    val FROG_MASK = "FROG_MASK".toInternalName()
+    private val frogMask by lazy { FROG_MASK.getItemStack() }
+
+    @HandleEvent(GuiRenderEvent.GuiOverlayRenderEvent::class, onlyOnIsland = IslandType.THE_PARK)
+    fun onRenderOverlay() {
+        if (!isEnabled()) return
+
+        config.displayPosition.renderRenderable(display, posLabel = "Frog Mask Display")
+    }
+
+    @HandleEvent(SecondPassedEvent::class, onlyOnIsland = IslandType.THE_PARK)
+    fun onSecondPassed() {
+        if (!isEnabled()) return
+        display = null
+
+        val helmet = InventoryUtils.getHelmet() ?: return
+        if (helmet.getInternalName() != FROG_MASK) return
+
+        activeRegionPattern.firstMatcher(helmet.getLore()) {
+            val currentRegion = group("region")
```

Comment:
rename to helmetRegion to differentiate between "region the player is currently at"